lib/ca_testing/patches/selenium_options.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module CaTesting
+  module Patches
+    class SeleniumOptions < Base
+      def initialize(browser)
+        @browser = browser
+        super()
+      end
+
+      def patch!
+        return unless valid?
+
+        super
+      end
+
+      def valid?
+        %i[firefox safari].include?(@browser)
+      end
+
+      private
+
+      def description
+        <<~DESCRIPTION
+          This patch fixes an issue with Selenium4 not camelising the browser_name property
+          The issue is the driver, which is now fully W3C conformant expects `browserName`
+
+          See: https://github.com/SeleniumHQ/selenium/pull/8834 for more details/discussion including this fix
+          LH - Nov 2020
+        DESCRIPTION
+      end
+
+      def perform
+        case @browser
+        when :firefox then Selenium::WebDriver::Firefox::Options.include CapabilitiesAsJsonFix
+        when :safari then Selenium::WebDriver::Safari::Options.include CapabilitiesAsJsonFix
+        end
+      end
+
+      def deprecation_notice_date
+        Time.new(2021, 12, 25)
+      end
+
+      def prevent_usage_date
+        Time.new(2022, 6, 30)
```

Comment:
FYI - This is fixed in the latest beta1 (But we're not upgrading yet). So this seems a realistic timeframe for them to ship v4 proper.